model/commitqueue/commit_queue.go

```diff
@@ -170,57 +170,57 @@ func ClearAllCommitQueues() (int, error) {
 	return clearedCount, nil
 }
 
-func RemoveCommitQueueItemForVersion(projectId, patchType, version string, user string) (bool, error) {
+func RemoveCommitQueueItemForVersion(projectId, version string, user string) (*CommitQueueItem, error) {
 	cq, err := FindOneId(projectId)
 	if err != nil {
-		return false, errors.Wrapf(err, "can't get commit queue for id '%s'", projectId)
+		return nil, errors.Wrapf(err, "can't get commit queue for id '%s'", projectId)
 	}
 	if cq == nil {
-		return false, errors.Errorf("no commit queue found for '%s'", projectId)
+		return nil, errors.Errorf("no commit queue found for '%s'", projectId)
 	}
 
-	issue := version
-	if patchType == SourcePullRequest {
-		head, valid := cq.Next()
-		// version is populated for PR items at the top of the queue only,
-		// so if the version for the item at the top of the queue doesn't match
-		// then the version is not here
-		if !valid || head.Version != version {
-			return false, nil
+	foundVersion := false
+	issue := ""
+	for _, item := range cq.Queue {
+		if item.Version == version {
+			foundVersion = true
+			issue = item.Issue
 		}
-		issue = head.Issue
+	}
+	if !foundVersion {
```

Comment:
Shouldn't we just call RemoveItemAndPreventMerge with "versionExists" set to false? Even if there's no version, we should still remove the item from the queue. Also, do we now set version for diff items? Before I believe this was only used for PRs.

Reply:
We shouldn't because not finding a version also means that we also have no issue. Actually foundVersion is redundant here, I removed it